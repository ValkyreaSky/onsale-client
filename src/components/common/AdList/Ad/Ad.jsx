import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
// Components
import { Card, Icon, Col } from 'antd';

const { Meta } = Card;

const AdContainer = styled(Card)`
  margin-bottom: 2rem !important;
`;

const AdImageContainer = styled.div`
  height: 15rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #e8e8e8;
  color: red;
`;

const AdImage = styled.img`
  display: block;
  width: auto;
  max-width: 100%;
  height: auto;
  max-height: 15rem;
`;

const Ad = ({
  ad: {
    _id, title, image, price,
  }, withRemove, handleRemove,
}) => {
  const imageResized = image.replace(/\/upload\//, '/upload/h_150/');

  return (
    <Col
      xs={{
        span: 24,
      }}
      sm={{
        span: 12,
      }}
      md={{
        span: 6,
      }}
      lg={{
        span: 4,
      }}
    >
      <AdContainer
        hoverable
        actions={withRemove && [<Icon type="delete" onClick={handleRemove(_id)} theme="twoTone" twoToneColor="red" />, <Icon type="edit" />]}
        cover={(
          <Link to={`/ad/${_id}`}>
            <AdImageContainer>
              <AdImage data-src={imageResized} src="https://res.cloudinary.com/selli/image/upload/h_150/placeholder.png" alt={title} className="lazyload" />
            </AdImageContainer>
          </Link>
        )}
      >
        <Link to={`/ad/${_id}`}>
          <Meta
            title={title}
            description={`$${price}`}
          />
        </Link>
      </AdContainer>
    </Col>
  );
};

Ad.defaultProps = {
  withRemove: false,
  handleRemove: () => {},
};

Ad.propTypes = {
  ad: PropTypes.objectOf(PropTypes.any).isRequired,
  withRemove: PropTypes.bool,
  handleRemove: PropTypes.func,
};

export default Ad;
