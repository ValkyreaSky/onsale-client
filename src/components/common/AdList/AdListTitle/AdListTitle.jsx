import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Title = styled.h2`
  margin: 2rem 0;
  font-size: 2.2rem;
`;

const SmallTitle = styled(Title)`
  font-size: 1.7rem;
`;

const AdListTitle = ({ text, small }) => {
  if (small) {
    return <SmallTitle>{text}</SmallTitle>;
  }
  return <Title>{text}</Title>;
};

AdListTitle.defaultProps = {
  small: false,
};

AdListTitle.propTypes = {
  text: PropTypes.string.isRequired,
  small: PropTypes.bool,
};

export { Title, SmallTitle };
export default AdListTitle;
