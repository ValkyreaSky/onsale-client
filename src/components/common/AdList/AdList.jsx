import React from 'react';
import PropTypes from 'prop-types';
// Components
import { Row } from 'antd';
import Ad from 'components/common/AdList/Ad/Ad';
import AdListTitle from 'components/common/AdList/AdListTitle/AdListTitle';

const AdList = ({ ads, withRemove, handleRemove }) => {
  let render;

  if (ads.length === 0) {
    render = <AdListTitle small text="Sorry, we didn't find any results." />;
  } else {
    render = (
      <Row gutter={16}>
        {ads.map(ad => <Ad withRemove={withRemove} handleRemove={handleRemove} key={ad._id} ad={ad} />)}
      </Row>
    );
  }

  return render;
};

AdList.defaultProps = {
  withRemove: false,
  handleRemove: () => {},
};

AdList.propTypes = {
  withRemove: PropTypes.bool,
  handleRemove: PropTypes.func,
  ads: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default AdList;
