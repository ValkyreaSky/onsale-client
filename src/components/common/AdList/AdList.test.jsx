import React from 'react';
import { shallow } from 'enzyme';
import AdList from './AdList';
import Ad from './Ad/Ad';
import AdListTitle from './AdListTitle/AdListTitle';

describe('<AdListTitle />', () => {
  let wrapperWithAds; let
    wrapperWithoutAds;

  const ads = [
    {
      _id: 1,
    },
    {
      _id: 2,
    },
  ];

  beforeAll(() => {
    wrapperWithAds = shallow(<AdList ads={ads} />);
    wrapperWithoutAds = shallow(<AdList ads={[]} />);
  });

  it('should match snapthot', () => {
    expect(wrapperWithAds).toMatchSnapshot();
    expect(wrapperWithoutAds).toMatchSnapshot();
  });

  it('should render <Ad> components when "ads" prop is not empty', () => {
    expect(wrapperWithAds.find(Ad).length).toBe(2);
  });

  it('should not render <Ad> components when "ads" prop is empty', () => {
    expect(wrapperWithoutAds.find(Ad).length).toBe(0);
  });

  it('should render <AdListTitle> component when "ads" prop is empty', () => {
    expect(wrapperWithoutAds.find(AdListTitle).length).toBe(1);
  });
});
