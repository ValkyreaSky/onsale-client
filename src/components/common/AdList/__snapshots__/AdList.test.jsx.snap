// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdListTitle /> should match snapthot 1`] = `
<Row
  gutter={16}
>
  <Ad
    ad={
      Object {
        "_id": 1,
      }
    }
    handleRemove={[Function]}
    key="1"
    withRemove={false}
  />
  <Ad
    ad={
      Object {
        "_id": 2,
      }
    }
    handleRemove={[Function]}
    key="2"
    withRemove={false}
  />
</Row>
`;

exports[`<AdListTitle /> should match snapthot 2`] = `
<AdListTitle
  small={true}
  text="Sorry, we didn't find any results."
/>
`;
